### ISSUE TITLE
Complex number support for pinv

### ISSUE THREAD

**Issue by themightyoarfish** [cupy/cupy#1615]
Currently, `cupy.linalg.pinv` does not seem to work with complex numbers.
This code
```
u, s, vt = decomposition.svd(a, full_matrices=False)
cutoff = rcond * s.max()     # error
s1 = 1 / s
s1[s <= cutoff] = 0    # error
return core.dot(vt.T, s1[:, None] * u.T)
``` 
does not work since `max()` is not supported on complex arrays (see #794) and the same holds for comparison operators.

**Comment by bamartin125** [comment]
Looking forward to having this implemented. @themightyoarfish, I'll likely start testing your patch tomorrow. Please let me know if there is any other ways I can help.

**Comment by leofang** [comment]
I suppose this is closed by #1636.

**Comment by carterbox** [comment]
This may have been resolved when #2581 was resolved.

**Comment by leofang** [comment]
@kmaehashi This can be closed.
